api_client/python/timesketch_api_client/search.py

```diff
@@ -0,0 +1,840 @@
+                    response, message='Unable to query results',
+                    error=ValueError)
+            more_response_json = error.get_response_json(more_response, logger)
+            count = len(more_response_json.get('objects', []))
+            total_count += count
+            response_json['objects'].extend(
+                more_response_json.get('objects', []))
+            more_meta = more_response_json.get('meta', {})
+            added_time = more_meta.get('es_time', 0)
+            response_json['meta']['es_time'] += added_time
+
+        total_elastic_count = response_json.get(
+            'meta', {}).get('es_total_count', 0)
+        if total_elastic_count != total_count:
+            logger.info(
+                '%d results were returned, but '
+                '%d records matched the search query',
+                total_count, total_elastic_count)
+
+        self._raw_response = response_json
+
+    def add_chip(self, chip):
+        """Add a chip to the ..."""
+        self._chips.append(chip)
+        self.commit()
+
+    def add_date_range(self, start_time, end_time):
+        """Add a date range chip to the search query.
+
+        Args:
+            start_time (str): a string with the start time of the range,
+                the format should be '%Y-%m-%dT%H:%M:%S'
+            end_time (str): a string with the end time of the range,
+                the format should be '%Y-%m-%dT%H:%M:%S'
+        """
+        chip = DateRangeChip()
+        chip.start_time = start_time
+        chip.end_time = end_time
+        self.add_chip(chip)
+
+    @property
+    def chips(self):
+        """Property that returns all the chips in the search object."""
+        return self._chips
+
+    def commit(self):
+        """Commit changes to the search object."""
+        self._raw_response = None
+        super().commit()
+
+    @property
+    def created_at(self):
+        """Property that returns back the creation time of a search."""
+        return self._created_at
+
+    def delete(self):
+        """Deletes the saved search from the store."""
+        if not self._resource_id:
+            logger.warning(
+                'Unable to delete the saved search, it does not appear to be '
+                'saved in the first place.')
+            return False
+
+        resource_url = (
+            f'{self.api.api_root}/sketches/{self._sketch.id}/views/'
+            f'{self._resource_id}/')
+        response = self.api.session.delete(resource_url)
+        return error.check_return_status(response, logger)
+
+    @property
+    def description(self):
+        """Property that returns back the description of the saved search."""
+        return self._description
+
+    @description.setter
+    def description(self, description):
+        """Make changes to the saved search description field."""
+        self._description = description
+        self.commit()
+
+    def from_explore(  # pylint: disable=arguments-differ
```

Comment:
from_explore? What does that mean? shouldn't this just be `search()` ?

Reply:
well, this is coming from the base object as well... to be consistent... we can name it something different, but we've already got the aggregation object defining that function name... 

we can just drop this also from the base class and rename this one

Reply:
changed to from_manual() as discussed offline